src/main/java/seedu/duke/parsers/ModHappyParser.java

```diff
@@ -56,4 +50,26 @@ public Command parseCommand(String userInput) throws ModHappyException {
         }
     }
 
+    /**
+     * Returns the Parser object for parsing commands associated with the given command word.
+     * If the command takes no arguments, null is returned.
+     * @param commandWord the command word (e.g. "add")
+     * @return an instance of the relevant Parser object or null
+     * @throws UnknownCommandException if commandWord does not correspond with any command
+     */
+    private Parser getCommandParser(String commandWord) throws UnknownCommandException {
+        switch (commandWord) {
+        case (EXIT_COMMAND_WORD):
+        case (LIST_COMMAND_WORD):
+            // Intentional fallthrough
+            return new NoArgumentParser(commandWord);
```

Comment:
No command parser given for EXIT_COMMAND_WORD. Also, because not all commands need parser actually. For example, exit and help may tasks no parameters. In this case, a wrapper parser doing nothing seems meaningless. May consider. I suggest to put the switch in the  "parseCommand(String userInput)" and use parser whenever needed in certain case. For example. when it's figured as the "add", which needs parser, and you can just use the AddParser() in that particular case.

Reply:
NoArgumentParser ensures that commands that take no arguments (like "exit") will not run if unnecessary arguments are provided (e.g. "exit hello"), since other teams might report this behaviour as a bug.